.ci/pytorch/test.sh

```diff
@@ -400,7 +400,15 @@ pr_time_benchmarks() {
   source benchmarks/dynamo/pr_time_benchmarks/benchmark_runner.sh "$TEST_REPORTS_DIR/pr_time_benchmarks_after.txt" "benchmarks/dynamo/pr_time_benchmarks/benchmarks"
   echo "benchmark results on current PR: "
   cat  "$TEST_REPORTS_DIR/pr_time_benchmarks_after.txt"
-
+  git checkout HEAD~1
```

Comment:
Do we need to worry about the fact that we're leaving it checked out at HEAD~1?  Maybe restore it before exiting the function.

Reply:
What if there are more than 1 commit in the PR?
You should look at `test_forward_backward_compatibility` and probably do the same.

Reply:
@aorenste I do exit after i run the pr_time_benchmarks tests, so nothing after that run, no need to restore.
@kit1980 I can check and try to rebuild same way as in test_forward_backward_compatibility, maybe i can move that to  a shared function